File: ConferenceRooms.Api/Models/Room.cs
namespace ConferenceRooms.Api.Models;

public class Room
{
    public int Id { get; set; }
    public required string Name { get; set; }
    public int Capacity { get; set; }
    public decimal BasePricePerHour { get; set; }
    public ICollection<RoomService> Services { get; set; } = [];
    public ICollection<Booking> Bookings { get; set; } = [];
}

File: ConferenceRooms.Admin/src/app/core/models/room.ts
export interface RoomService {
  id: number;
  name: string;
  price: number;
}

export interface Room {
  id: number;
  name: string;
  capacity: number;
  basePricePerHour: number;
  services: RoomService[];
}

export interface CreateRoomServiceRequest {
  name: string;
  price: number;
}

export interface CreateRoomRequest {
  name: string;
  capacity: number;
  basePricePerHour: number;
  services?: CreateRoomServiceRequest[];
}

export interface UpdateRoomRequest {
  name: string;
  capacity: number;
  basePricePerHour: number;
}
